pkg/source/archive.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright 2022 VMware, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package source
+
+import (
+	"archive/zip"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+)
+
+func HandleZip(fileName string) (string, error) {
+	tmpDir, err := ioutil.TempDir("", "")
```

Comment:
issue: When is the temp directory removed?
```
	tmpDir, err := ioutil.TempDir("", "")
```

Reply:
So we need the temp dir for imgpkg to package up the contents. So in the called function, tmp dir is cleanedup.

Reply:
it would be better to accept the directory as an argument, and then the cleanup is clearly the callers problem